x-pack/plugins/security_solution/server/lib/detection_engine/signals/send_telemetry_events.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { TelemetryEventsSender, TelemetryEvent } from '../../telemetry/sender';
+import { RuleTypeParams } from '../types';
+import { BuildRuleMessage } from './rule_messages';
+import { SignalSearchResponse, SignalSource } from './types';
+import { Logger } from '../../../../../../../src/core/server';
+
+export interface SearchResultWithSource {
+  _source: SignalSource;
+}
+
+export function selectEvents(filteredEvents: SignalSearchResponse): TelemetryEvent[] {
+  const sources = filteredEvents.hits.hits.map(function (
+    obj: SearchResultWithSource
+  ): TelemetryEvent {
+    return obj._source;
+  });
+
+  return sources.filter(function (obj: TelemetryEvent) {
+    // Filter out non-endpoint alerts
+    return obj.datastream?.dataset === 'endpoint.alerts';
+  });
```

Comment:
Can shorthand with an arrow function if you'd like:

```suggestion
  // Filter out non-endpoint alerts
  return sources.filter((obj: TelemetryEvent) => obj.datastream?.dataset === 'endpoint.alerts');
```